x/oracle/keeper.go

```diff
@@ -0,0 +1,364 @@
+func (k Keeper) DeleteMissCounter(ctx sdk.Context, operator sdk.ValAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.GetMissCounterKey(operator))
+}
+
+// IterateMissCounters iterates over the miss counters and performs a callback function.
+func (k Keeper) IterateMissCounters(ctx sdk.Context,
+	handler func(operator sdk.ValAddress, missCounter uint64) (stop bool)) {
+
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, types.MissCounterKey)
+	defer iter.Close()
+	for ; iter.Valid(); iter.Next() {
+		operator := sdk.ValAddress(iter.Key()[2:])
+
+		var missCounter gogotypes.UInt64Value
+		k.cdc.MustUnmarshal(iter.Value(), &missCounter)
+
+		if handler(operator, missCounter.Value) {
+			break
+		}
+	}
+}
+
+//-----------------------------------
+// AggregateExchangeRatePrevote logic
+
+// GetAggregateExchangeRatePrevote retrieves an oracle prevote from the store
+func (k Keeper) GetAggregateExchangeRatePrevote(ctx sdk.Context,
+	voter sdk.ValAddress) (aggregatePrevote types.AggregateExchangeRatePrevote,
+	err error) {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(types.GetAggregateExchangeRatePrevoteKey(voter))
+	if b == nil {
+		err = sdkerrors.Wrap(types.ErrNoAggregatePrevote, voter.String())
+		return
+	}
+	k.cdc.MustUnmarshal(b, &aggregatePrevote)
+	return
+}
+
+// SetAggregateExchangeRatePrevote set an oracle aggregate prevote to the store
+func (k Keeper) SetAggregateExchangeRatePrevote(ctx sdk.Context,
+	voter sdk.ValAddress,
+	prevote types.AggregateExchangeRatePrevote) {
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshal(&prevote)
+
+	store.Set(types.GetAggregateExchangeRatePrevoteKey(voter), bz)
+}
+
+// DeleteAggregateExchangeRatePrevote deletes an oracle prevote from the store
+func (k Keeper) DeleteAggregateExchangeRatePrevote(ctx sdk.Context, voter sdk.ValAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.GetAggregateExchangeRatePrevoteKey(voter))
+}
+
+// IterateAggregateExchangeRatePrevotes iterates rate over prevotes in the store
+func (k Keeper) IterateAggregateExchangeRatePrevotes(ctx sdk.Context,
+	handler func(voterAddr sdk.ValAddress,
+		aggregatePrevote types.AggregateExchangeRatePrevote) (stop bool)) {
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, types.AggregateExchangeRatePrevoteKey)
+	defer iter.Close()
+	for ; iter.Valid(); iter.Next() {
+		voterAddr := sdk.ValAddress(iter.Key()[2:])
+
+		var aggregatePrevote types.AggregateExchangeRatePrevote
+		k.cdc.MustUnmarshal(iter.Value(), &aggregatePrevote)
+		if handler(voterAddr, aggregatePrevote) {
+			break
+		}
+	}
+}
+
+//-----------------------------------
+// AggregateExchangeRateVote logic
+
+// GetAggregateExchangeRateVote retrieves an oracle prevote from the store
+func (k Keeper) GetAggregateExchangeRateVote(ctx sdk.Context,
+	voter sdk.ValAddress) (aggregateVote types.AggregateExchangeRateVote,
+	err error) {
```

Comment:
nit: please cleanup formatting here.

Reply:
made changes, will let you resolve if you are good with proposed formatting

Reply:
I would remove the named return values to shorten the signature.